package/lib/compileFunctions.js

```diff
@@ -145,12 +159,39 @@ const validateEventsProperty = (funcObject, functionName) => {
 
 const validateVpcConnectorProperty = (funcObject, functionName) => {
   if (funcObject.vpc && typeof funcObject.vpc === 'string') {
-    const vpcNamePattern = /projects\/[\s\S]*\/locations\/[\s\S]*\/connectors\/[\s\S]*/i;
-    if (!vpcNamePattern.test(funcObject.vpc)) {
+    // vpcConnector argument can be one of two possible formats as described here:
+    // https://cloud.google.com/functions/docs/reference/rest/v1/projects.locations.functions#resource:-cloudfunction
+    if (funcObject.vpc.indexOf('/') > -1) {
+      const vpcNamePattern = /projects\/[\s\S]*\/locations\/[\s\S]*\/connectors\/[\s\S]*/i;
+      if (!vpcNamePattern.test(funcObject.vpc)) {
+        const errorMessage = [
+          `The function "${functionName}" has invalid vpc connection name`,
+          ' VPC Connector name should follow projects/{project_id}/locations/{region}/connectors/{connector_name}',
+          ' or just {connector_name} if within the same project.',
+          ' Please check the docs for more info at ',
+          ' https://cloud.google.com/functions/docs/reference/rest/v1/projects.locations.functions#resource:-cloudfunction',
+        ].join('');
+        throw new Error(errorMessage);
+      }
+    }
+  }
+};
+
+const validateVpcConnectorEgressProperty = (funcObject, functionName) => {
+  if (funcObject.vpcEgress && typeof funcObject.vpcEgress === 'string') {
```

Comment:
This new property should be added to configuration schema in `provider/googleProvider.js`

Reply:
hey! how are you? sorry, I was trying to make these changes since these would be of great help to me.
I can't find where to add that property in that file. Would you help me?

Thank you!

Reply:
Hello :wave: The schema for provider and function properties is here: https://github.com/serverless/serverless-google-cloudfunctions/blob/e6e079965eb5303cb5a33f549f56eb8c5f981688/provider/googleProvider.js#L103-L129